### ISSUE TITLE
table-list labels are turned into readonly inputs instead of group labels

### ISSUE THREAD

**Issue by MartijnR** [SEL-Columbia/pyxform#72]
Not sure if this by design:

table-list outputs:

``` xml
<group>
    <label></label>
    <input><!-- a readonly input -->
         <label>the label of the table</label>
    </input>
    ....
```

instead of:

``` xml
<group>
    <label>the label of the table</label>
    .....
```

The problem is that group label looks visually different form a readonly input label and making them look the same seems a bit of a hack.

**Comment by nathanathan** [comment]
I don't think it's a hack. table-lists already involve generating some extra xml for the table heading and group labels render too small in Collect and I'm not sure if they can include hints/media.

The small text at the top is the group label:
![group label](http://opendatakit.org/wp-content/uploads/static/form-design/images-scaled/multichoicelist.png)

On further thought, it does seem like a bit of a hack, but I still think we should keep it.

**Comment by MartijnR** [comment]
Just to be sure, we're talking about the same. I'm referring to the label of the group itself (not the label of the column headings). Shouldn't this output the same XML for the label as it does when you don't use the table-list appearance attribute but use the more verbose method instead?

![Screen Shot 2013-02-08 at 12 21 59 PM](https://f.cloud.github.com/assets/627350/140568/e9824ac4-7224-11e2-83f4-05514c66a02f.png)

**Comment by MartijnR** [comment]
So in the above image should _'List Group'_ use the XML syntax as _'Table (made with an easier method)'_

**Comment by nathanathan** [comment]
Not necessarily. In your example table-list is a short hand for adding an extra appearance=label row.
Why not have it be short hand for adding an extra note row and an extra appearance=label row?

i.e.

| type | name | label | appearance |
| --- | --- | --- | --- |
| begin_group |  | head | table-list |
| select_one | q1 | Q1 |  |
| end_group |  |  |  |

=>

| type | name | label | appearance |
| --- | --- | --- | --- |
| begin_group |  |  | field-list |
| note |  | head |  |
| select_one |  |  | label |
| select_one | q1 | Q1 |  |
| end_group |  |  |  |

**Comment by MartijnR** [comment]
hmmm.... I respectfully disagree. :)

**Comment by nathanathan** [comment]
Do you disagree because you think table-lists should have the smaller labels or because of a back-end issue?
How about if we resolve this by getting a third opinion. @ukanga or @prabhasp do either of you have a preference on the matter?

**Comment by MartijnR** [comment]
I think we should use `<label>` for what it is meant for. So I disagree for semantic reasons with using notes as a label, and leaving the label element empty. Doing this should be assumed to have consequences in the way the 'label' appears in a client even though it may look fine in ODK Collect. ( I also have a prejudice against readonly inputs in XForms in general... but that's not so relevant in this case ;).)

**Comment by prabhasp** [comment]
My opinion: purely, what Martijn is saying is right; it seems much more
semantically correct to do this as a group label. ODK collect doesn't do
this, so I think the question gets stuck there. Even if we wanted it to,
because ODK is screen-by-screen, the display of group labels as big, or
even at a similar font size etc. as question text is pretty much
impossible. So, I think we're stuck with historical reasons to go the label
route.

The other thing we could do is to do table-list-single-screen
table-list-multi-screen or something like that, but that revolts me even
more from a user perspective.

On Wed, Feb 13, 2013 at 10:45 AM, Martijn van de Rijdt <
notifications@github.com> wrote:

> I think we should use <label> for what it is meant for. So I disagree for
> semantic reasons with using notes as a label, and leaving the label element
> empty. Doing this should be assumed to have consequences in the way the
> 'label' appears in a client even though it may look fine in ODK Collect. (
> I also have prejudice against readonly inputs in XForms in general... but
> that's not so relevant in this case ;).)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/modilabs/pyxform/issues/72#issuecomment-13500166.

## 

Prabhas Pokharel
http://twitter.com/prabhasp
US mobile: +1 347 948 7654
skype/facebook/whatever: prabhasp

**Comment by MartijnR** [comment]
@larryweya, anybody any interest or time to take this on....? 

cc. @prabhasp
